### ISSUE TITLE
[Bug]Landscape Kodi 19

### ISSUE THREAD

**Issue by Blurayx** [jurialmunkey/plugin.video.themoviedb.helper#446]
Kodi 19
I mirrored all the settings from Kodi 18. Unfortunately, landscape has stopped working.

<img width="1412" alt="Screenshot 2021-02-24 at 18 35 32" src="https://user-images.githubusercontent.com/2873112/109063352-f62f9f00-76e0-11eb-9754-a8cb4e0f00ce.png">

<img width="1126" alt="Screenshot 2021-02-24 at 18 34 47" src="https://user-images.githubusercontent.com/2873112/109063424-05aee800-76e1-11eb-9e59-cec78a882e77.png">

Thanks

**Comment by Blurayx** [comment]
Update:

This is the same issues as before with Kodi 18
https://github.com/jurialmunkey/plugin.video.themoviedb.helper/issues/425

**Comment by jurialmunkey** [comment]
Are you sure you have landscape art in your library for the show?

![Honeyview_screenshot00006](https://user-images.githubusercontent.com/4618719/109404264-72173900-79b8-11eb-8191-ffd82e0f7ec7.jpg)


Because it is working perfectly fine for me:

With Landscape enabled:
![Honeyview_screenshot00007](https://user-images.githubusercontent.com/4618719/109404267-7e02fb00-79b8-11eb-8de0-23732e26f146.jpg)

With Fanart enabled:
![Honeyview_screenshot00008](https://user-images.githubusercontent.com/4618719/109404272-8d824400-79b8-11eb-9ddc-bf3abf6dccdb.jpg)

And this artwork is definitely coming from my library because it updates when I change it.

**Comment by jurialmunkey** [comment]
Also with this particular show there is no TMDb ID on Trakt for the show, so it is going to be impossible to match

![Screenshot 2021-02-28 114338](https://user-images.githubusercontent.com/4618719/109404447-54e36a00-79ba-11eb-9aa7-c4e766f7871e.jpg)

**Comment by Blurayx** [comment]
> Are you sure you have landscape art in your library for the show?
> 
> ![Honeyview_screenshot00006](https://user-images.githubusercontent.com/4618719/109404264-72173900-79b8-11eb-8191-ffd82e0f7ec7.jpg)
> 
> Because it is working perfectly fine for me:
> 
> With Landscape enabled:
> ![Honeyview_screenshot00007](https://user-images.githubusercontent.com/4618719/109404267-7e02fb00-79b8-11eb-8de0-23732e26f146.jpg)
> 
> With Fanart enabled:
> ![Honeyview_screenshot00008](https://user-images.githubusercontent.com/4618719/109404272-8d824400-79b8-11eb-9ddc-bf3abf6dccdb.jpg)
> 
> And this artwork is definitely coming from my library because it updates when I change it.

Yes, I created my own landscape for Nostradamus end of days and it works perfectly with Kodi 18

<img width="118" alt="Screenshot 2021-02-28 at 01 42 39" src="https://user-images.githubusercontent.com/2873112/109405362-c940ed80-7967-11eb-8cbf-22de05d42039.png">

**Comment by Blurayx** [comment]
Just to be clear I'm talking about Next Aired TV show widgets and Next Aired home screen widgets

**Comment by jurialmunkey** [comment]
Like I said, there's no TMDb ID on Trakt for this show, so it isn't possible to link.

Also, make sure you've actually scanned the landscape art to the Kodi library by checking that you have it under "Select Art" in the info dialog in the library. There were a lot of changes for artwork handling in Matrix.

**Comment by Blurayx** [comment]
Thanks for your help. No matter what I do nothing works.

Gone back to Kodi 18 because it works perfectly for me. As you can see no issues with widgets.

<img width="1595" alt="Screenshot 2021-02-28 at 08 41 53" src="https://user-images.githubusercontent.com/2873112/109412619-ddeda780-79a0-11eb-974e-c7327b3ed4d3.png">

**Comment by Blurayx** [comment]
btw is the poster view option for Next Aired Liberty?

**Comment by Blurayx** [comment]
Hi jurialmunkey
Update: Kodi 19 Landscape issues with Next Aired 

I decided to try Kodi 19 again. Can you please explain to me what's happening when I enable Thumb? 
Basically, I gain most of my landscape but lose the functionality of “show information for unwatched items” for episode thumb.

This doesn't affect my Next Aired landscape with Kodi 18. Hopefully, this will explain the issues I'm having with Kodi 19


https://user-images.githubusercontent.com/2873112/109669039-a1b97300-7b69-11eb-8a4c-39c1555a2b2d.mp4

Thanks

**Comment by jurialmunkey** [comment]
Ah I see, you are talking about with "Hide Episode Thumbs" enabled. This is due to a Matrix change in how Kodi hides spoiler thumbs.

To hide spoilers in Leia, Kodi core would replace the thumbnail image path in the `Art(Thumb)` field with the `OverlaySpoiler.png` icon path. There was a change in Matrix where now Kodi core instead replaces the thumbnail image with the fanart image directly. As a result it is possible for skins to check if a thumb was replaced to hide spoilers.

Basically, what used to happen in Leia was:

1. TMDbH fills the `Art(Thumb)` field with the filepath of the desired artwork type (thumb, fanart or landscape).
2. Kodi replaces the `Art(Thumb)` field with `OverlaySpoiler.png` if the episode is unwatched and hide spoilers enabled.
3. Arctic Horizon checks if the `Art(Thumb)` filename is`OverlaySpoiler.png` and instead shows `Art(Landscape)` if it is.

Now step 3 is impossible because there isn't a single filename to check against anymore since each fanart will have a different filename. Both TMDbH and Arctic Horizon are working correctly - it just isn't possible to check properly anymore.

**Comment by jurialmunkey** [comment]
I've added a workaround in most recent matrix version of Arctic Horizon on github. I'm now also comparing the thumb against the fanart image. Frustrating because I now need to do two checks instead of one to determine if thumb was changed due to spoilers - so costs more CPU cycles but hopefully performance impact shouldn't be too bad since it is an equality comparison which are generally fast.

**Comment by Blurayx** [comment]
Thank you so much for your patients and time in solving the bug. I thought I was going to be stuck with Leia because you weren't having the same problem. I haven't noticed any performance impact sofar.
I'm so impressed I sent a well-deserved donation!!! 😀

**Comment by Blurayx** [comment]
Hi Jurialmunkey.

Sorry to be a pain. Everything is working. Unfortunately when I select watched/unwatched, My landscape images changes.

https://user-images.githubusercontent.com/2873112/110213539-6f28b680-7e98-11eb-9e74-f0f71dbfe51f.mp4


Thanks

**Comment by jurialmunkey** [comment]
Found the issue. Kodi now overrides artwork for plugin items that have a DBID corresponding to an item in your library. So if the episode is in your library then Kodi will always override the artwork regardless of what TMDbH sets as the thumb.

You don't see it with the unwatched items because for those Kodi makes Thumb=Fanart and so the fix I made to AH catches those and swaps them out to landscape art at skin level. And I wasn't seeing it because all my test items hadn't aired yet and so weren't in my library. 

Anyway, added a quick and dirty workaround to drop the episode DBID for any calendar items when the artwork override option is enabled. The linkback action to open the library should still work because that uses the TvShow DBID (not episode).
